structuralcodes/codes/mc2010/_concrete_shear.py

```diff
@@ -0,0 +1,1598 @@
+                    Med,
+                    Ved,
+                    Ned,
+                    delta_e,
+                    alfa,
+                    gamma_c,
+                ),
+            )
+
+        if approx_lvl_s in (1, 2):
+            return min(
+                v_rds(asw, sw, z, f_ywd, theta, alfa),
+                v_rd_max(
+                    approx_lvl_s,
+                    fck,
+                    bw,
+                    theta,
+                    z,
+                    E_s,
+                    As,
+                    Med,
+                    Ved,
+                    Ned,
+                    delta_e,
+                    alfa,
+                    gamma_c,
+                ),
+            )
+    raise ValueError("Invalid approx level")
+
+
+def v_rdc(
+    approx_lvl_c: int,
+    approx_lvl_s: int,
+    fck: float,
+    z: float,
+    bw: float,
+    dg: float,
+    E_s: float,
+    As: float,
+    Med: float,
+    Ved: float,
+    Ned: float,
+    delta_e: float,
+    alfa: float,
+    gamma_c: float = 1.5,
+) -> float:
+    """The design shear resistance of a web or a slab without
+    shear reinforcement.
+
+    fib Model Code 2010, Eq. (7.3-17)
+
+    Args:
+        approx_lvl_c (int): Approximation level for concrete
+        approx_lvl_s (int): Approximation level for reinforcement
+        reinforcment (bool): Shear reinforced concrete or no shear
+        reinforcement
+        fck (float): Characteristic strength in MPa
+        z: (float): The length to the areasenter of cross-section in mm
+        bw: (float): Thickness of web in cross section in mm
+        dg: (float): Maximum size of aggregate
+        E_s: (float): The E_s-modulus to the materialb in MPa
+        As: (float): The cross-section area in mm^2
+        Med: (float): The moment working on the material in Nmm
+        Ved: (float): The shear working on the material in N
+        Ned: (float): The normal force working on the material in N
+        delta_e (float): The exentricity of the load in mm
+        alfa (float): Inclination of the stirrups in degrees
+        gamma_c (float): Safety factor
+
+    Returns:
+        float: The design shear resistance attributed to the concrete
+    """
+    if approx_lvl_s == 3:
+        return v_rdc_approx3(
+            approx_lvl_s,
+            fck,
+            z,
+            bw,
+            E_s,
+            As,
+            Med,
+            Ved,
+            Ned,
+            delta_e,
+            alfa,
+            gamma_c,
+        )
+    if approx_lvl_c == 1:
+        return v_rdc_approx1(fck, z, bw, gamma_c)
+
+    if approx_lvl_c == 2:
+        return v_rdc_approx2(
+            fck, z, bw, dg, E_s, As, Med, Ved, Ned, delta_e, gamma_c
+        )
+    raise ValueError("Invalid approx level")
+
+
+def v_rdc_approx1(
+    fck: float,
+    z: float,
+    bw: float,
+    gamma_c: float = 1.5,
+) -> float:
+    """For members with no segnificant axal load, with fyk <= 600 Mpa,
```

Comment:
Describe what the function does, e.g. `Shear capacity of concrete, Level I Approximation`, then followed by a blank line before the text `For members with ...`.